### ISSUE TITLE
Url_fopen

### ISSUE THREAD

**Issue by miyorineko** [flagrow/bazaar#69]
As you know, the allow_url_fopen is usually off due to security risks. Is it better using curl to fetch the file instead of url_fopen?

**Comment by luceos** [comment]
Thank you good point, I take it you're referencing [this line](https://github.com/flagrow/bazaar/blob/144f6b4b82853de498251309697a5979fc9185e7/src/Composer/Utils/ComposerFileEditor.php#L36) I see no other mention of `file_get_contents`.

**Comment by miyorineko** [comment]
@luceos Indeed, I found no "file_get_contents" or "fopen" function associate with the file "packages.json" in the code. But I just can't install any extension in the bazaar, and got some odd errors like this in the log:

[2017-07-18 08:31:17] production.ERROR: The "https://packagist.org/packages.json" file could not be downloaded: allow_url_fopen must be enabled in php.ini (https:// wrapper is disabled in the server configuration by allow_url_fopen=0
failed to open stream: no suitable wrapper could be found)

**Comment by luceos** [comment]
Behind the scenes Bazaar uses the Composer package which uses file_get_contents [for pretty much](https://github.com/composer/composer/issues/4740) [everything](https://github.com/composer/composer/search?utf8=%E2%9C%93&q=file_get_contents&type=).

I am not sure we can do anything about this without completely dropping the composer dependency. This is a lot of work though. I am going to keep this on our backlog as a reminder though.

**Comment by Legoman99573** [comment]
As of recent, CPanel has removed the allow_url_fopen option from Multi php.ini Editor and the server overrides to ``allow_url_fopen=0``.

RIP on shared hosting.

**Comment by luceos** [comment]
@Legoman99573 unless the shared host allows it for all sites. I can't imagine why you'd block it globally anyway..

**Comment by Legoman99573** [comment]
@luceos it is done by default on a clean install. It can be enabled /via WHM, but most shared hosts/resellers kept it blocked for the same reason this issue was opened for.
